File: benchmark/c/opl/opl_emu_opl_key_scale_atten/opl_emu_opl_key_scale_atten.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <string.h>

#define opl_max(a,b) (((a)>(b))?(a):(b))

uint32_t opl_emu_opl_key_scale_atten(uint32_t block, uint32_t fnum_4msb)
;
uint32_t opl_emu_opl_key_scale_atten(uint32_t block, uint32_t fnum_4msb)
{
	static uint8_t const fnum_to_atten[16] = { 0,24,32,37,40,43,45,47,48,50,51,52,53,54,55,56 };
	int32_t result = fnum_to_atten[fnum_4msb] - 8 * (block ^ 7);
	return opl_max(0, result);
}

File: translation_result/real_world_code/fail/c/opl/opl_emu_opl_key_scale_atten/gemini/Restart/opl_emu_opl_key_scale_atten.rs

const OPL_MAX: fn(i32, i32) -> i32 = |a, b| if a > b { a } else { b };

fn opl_emu_opl_key_scale_atten(block: i32, fnum_4msb: i32) -> i32 {
    static FNUM_TO_ATTEN: [i8; 16] = [
        0, 24, 32, 37, 40, 43, 45, 47, 48, 50, 51, 52, 53, 54, 55, 56,
    ];
    let result = FNUM_TO_ATTEN[fnum_4msb as usize] as i32 - 8 * (block ^ 7);
    OPL_MAX(0, result)
}
